src/smif/data_layer/memory_interface.py

```diff
@@ -75,9 +77,18 @@ def delete_sector_model(self, sector_model_name):
         del self._sector_models[sector_model_name]
     # endregion
 
+    # region Interventions
+    def read_interventions(self, sector_model_name):
+        return {item['name']: item for item in self._interventions[sector_model_name]}
```

Comment:
Avoid the comprehension for simplicity - `return self._interventions[sector_model_name]`